--- src/decompiler/heritage.cc	2021-06-03 19:24:14.000000000 +0200
+++ src/decompiler/heritage.cc	2021-06-04 10:57:56.000000000 +0200
@@ -1610,9 +1610,10 @@
   // Alter the disjoint cover (both locally and globally) to reflect our refinement
   LocationMap::iterator iter = disjoint.find(addr);
   int4 pass = (*iter).second.pass;
-  disjoint.erase(iter);
+  // disjoint.erase(iter);
   iter = globaldisjoint.find(addr);
-  globaldisjoint.erase(iter);
+  // globaldisjoint.erase(iter);
+
   Address curaddr = addr;
   int4 cut = 0;
   int4 intersect;
