--- a/src/decompiler/types.h
+++ b/src/decompiler/types.h
@@ -71,8 +71,12 @@ typedef char int1;
 typedef uint8 uintp;
 #endif
 
-#if defined (__linux__) && defined (__i386__)
+#if defined (__linux__) && (defined (__i386__) || defined (__arm__))
+#if defined(__BYTE_ORDER__) && __BYTE_ORDER__ == __ORDER_BIG_ENDIAN__
+#define HOST_ENDIAN 1
+#else
 #define HOST_ENDIAN 0
+#endif
 typedef unsigned long uintm;
 typedef long intm;
 typedef unsigned long long uint8;
@@ -86,8 +90,12 @@ typedef char int1;
 typedef uint4 uintp;
 #endif
 
-#if defined (__linux__) && defined (__x86_64__)
+#if defined (__linux__) && (defined (__x86_64__) || defined (__aarch64__))
+#if defined(__BYTE_ORDER__) && __BYTE_ORDER__ == __ORDER_BIG_ENDIAN__
+#define HOST_ENDIAN 1
+#else
 #define HOST_ENDIAN 0
+#endif
 typedef unsigned int uintm;
 typedef int intm;
 typedef unsigned long uint8;
-- 
2.24.3 (Apple Git-128)

