diff --git a/src/decompiler/types.h b/src/decompiler/types.h
index ff6f423ab..6d5b6acb2 100644
--- a/src/decompiler/types.h
+++ b/src/decompiler/types.h
@@ -71,7 +71,7 @@ typedef char int1;
 typedef uint8 uintp;
 #endif
 
-#if defined (__linux__) && (defined (__i386__) || defined (__arm__))
+#if (defined (__linux__) || defined (__FreeBSD__) || defined (__OpenBSD__)) && (defined (__i386__) || defined (__arm__))
 #if defined(__BYTE_ORDER__) && __BYTE_ORDER__ == __ORDER_BIG_ENDIAN__
 #define HOST_ENDIAN 1
 #else
@@ -90,7 +90,7 @@ typedef char int1;
 typedef uint4 uintp;
 #endif
 
-#if defined (__linux__) && (defined (__x86_64__) || defined (__aarch64__))
+#if (defined (__linux__) || defined (__FreeBSD__) || defined (__OpenBSD__)) && (defined (__x86_64__) || defined (__aarch64__))
 #if defined(__BYTE_ORDER__) && __BYTE_ORDER__ == __ORDER_BIG_ENDIAN__
 #define HOST_ENDIAN 1
 #else
