diff --git a/src/decompiler/database.cc b/src/decompiler/database.cc
index 26b68a6..ff2ad05 100644
--- a/src/decompiler/database.cc
+++ b/src/decompiler/database.cc
@@ -1958,7 +1958,8 @@ void ScopeInternal::addMapInternal(Symbol *sym,MapEntry *entry)
     string msg = "Symbol ";
     msg += sym->getName();
     msg += " extends beyond the end of the address space";
-    throw LowlevelError(msg);
+    fprintf (stderr, "%s\n", msg.c_str());
+    //throw LowlevelError(msg);
   }
     
   entry->mapIterator = rangemap->insert(entry,entry->addr.getOffset(),lastaddress.getOffset());
