diff --git a/src/decompiler/type.hh b/src/decompiler/type.hh
index 4289261..f90c216 100644
--- a/src/decompiler/type.hh
+++ b/src/decompiler/type.hh
@@ -119,10 +119,10 @@ protected:
   void saveXmlBasic(type_metatype meta,ostream &s) const;	///< Save basic data-type properties
   void saveXmlTypedef(ostream &s) const;	///< Write \b this as a \e typedef tag to stream
   void markComplete(void) { flags &= ~(uint4)type_incomplete; }		///< Mark \b this data-type as completely defined
-  virtual Datatype *clone(void) const=0;	///< Clone the data-type
   static uint8 hashName(const string &nm);	///< Produce a data-type id by hashing the type name
   static uint8 hashSize(uint8 id,int4 size);	///< Reversibly hash size into id
 public:
+  virtual Datatype *clone(void) const=0;	///< Clone the data-type
   /// Construct the base data-type copying low-level properties of another
   Datatype(const Datatype &op) { size = op.size; name=op.name; metatype=op.metatype; submeta=op.submeta; flags=op.flags;
     id=op.id; typedefImm=op.typedefImm; }
